Changes to support multiple cloud services from local WebUi

MMaiero opened the pull request:
This PR will add support, in the local WebUi, to multiple cloud services. Currently the implementation is fully working for the default stack. For Camel I've only updated versions and added some placeholders that eventually will need a full implementation to be used with the new setup.
Eventually, I can do a rollback of those changes if Camel cloud support will be not managed by this new feature.

Signed-off-by: MMaiero <matteo.maiero@eurotech.com>

cdealti commented:
@ctron Can you kindly review? See also #515 for the requirements and #539 for the proposed implementation.
If you agree this PR would be a good chance to finalize #189.

cdealti commented:
@ctron it's important to merge this before rewriting the history. Can you please review?

ctron commented:
I already started and did not get any response from @MMaiero 

Or maybe I did something wrong the new workflow for reviewing.

ctron commented:
But I don't completely understand the rush. Why can't we re-base afterwards?

cdealti commented:
I'd like to merge what we can before rewriting the history.

ctron commented:
Ok then I would suggest to fix the comments I made and proceed with merging. Anything which goes wrong can be fixed at a later time.

MMaiero commented:
I had no notifications till today when comments were made.
However I've updated the pull request with the suggested changes and rebased on develop.

ctron commented on kura/org.eclipse.kura.api/src/main/java/org/eclipse/kura/cloud/factory/CloudServiceFactory.java:
```diff
@@ -115,6 +124,8 @@
      */
     void createConfiguration(String pid) throws KuraException;
 
```
Add documentation

ctron replied:
And request from callers to handle a "null" result

ctron commented:
Ok, I see. Sorry fur that. Seems like you need to "submit" the comments on the new GitHub review process in order to make them visible.
